src/realm/obj.cpp

```diff
@@ -1492,7 +1483,8 @@ Obj& Obj::set<ObjLink>(ColKey col_key, ObjLink target_link, bool is_default)
     ColKey::Idx col_ndx = col_key.get_index();
     ColumnType type = col_key.get_type();
     if (type != ColumnTypeTraits<ObjLink>::column_id)
-        throw LogicError(LogicError::illegal_type);
+        throw InvalidArgument(ErrorCodes::TypeMismatch,
+                              util::format("Property not a %1", ColumnTypeTraits<int64_t>::column_id));
```

Comment:
I think this ObjLink code is dead, but anyhow it shouldn't be the int64_t type printed

Reply:
Fixed